macros/multiple_databases/timestamp_from_date.sql

```diff
@@ -1,18 +1,37 @@
 {%- macro timestamp_from_date(date_field) -%}
 
 {%- if target.type == 'snowflake' -%}
-    timestamp_from_parts({{ date_field }}, '0')
+    case
+        when len({{ date_field }}) > 0
+            then timestamp_from_parts({{ date_field }}, '0')
+        else
+            timestamp_from_parts(NULL, NULL)
+    end
 {%- elif target.type == 'sqlserver' -%}
-    datetime2fromparts(
-        datepart(year, {{ date_field }}),
-        datepart(month, {{ date_field }}),
-        datepart(day, {{ date_field }}),
-        0,
-        0,
-        0,
-        0,
-        3
-    )
+    case
+        when len({{ date_field }}) > 0
+            then datetime2fromparts(
+                datepart(year, {{ date_field }}),
+                datepart(month, {{ date_field }}),
+                datepart(day, {{ date_field }}),
+                0,
+                0,
+                0,
+                0,
+                3
+                )
+        else
+            datetime2fromparts(
+            datepart(year, NULL),
+            datepart(month, NULL),
+            datepart(day, NULL),
+            datepart(hour, NULL),
+            datepart(minute, NULL),
+            datepart(second, NULL),
+            datepart(millisecond, NULL),
+            3
+            )
```

Comment:
This can use the same simplification as the other macro. Also look at the indenting.